cmd/symbols/squirrel/lang_starlark.go

```diff
@@ -0,0 +1,184 @@
+package squirrel
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	sitter "github.com/smacker/go-tree-sitter"
+
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+func (squirrel *SquirrelService) getDefStarlark(ctx context.Context, node Node) (ret *Node, err error) {
+	defer squirrel.onCall(node, String(node.Type()), lazyNodeStringer(&ret))()
+	switch node.Type() {
+	case "identifier":
+		return squirrel.getDefStarlarkIdentifier(ctx, node)
+	case "string":
+		return squirrel.getDefStarlarkString(ctx, node)
+	default:
+		return nil, nil
+
+	}
+}
+
+func (squirrel *SquirrelService) getDefStarlarkIdentifier(ctx context.Context, node Node) (ret *Node, err error) {
+	symbol := node.Node.Content(node.Contents)
+	root := Node{
+		RepoCommitPath: node.RepoCommitPath,
+		Node:           getRoot(node.Node),
+		Contents:       node.Contents,
+		LangSpec:       node.LangSpec,
+	}
+	exports, err := starlarkBindingNamed(symbol, root)
+	if err != nil {
+		return nil, err
+	}
+	if exports != nil {
+		return &Node{
+			RepoCommitPath: node.RepoCommitPath,
+			Node:           exports,
+			Contents:       node.Contents,
+			LangSpec:       node.LangSpec,
+		}, nil
+	}
+	fmt.Println(node)
+	return nil, nil
+}
+
+func (squirrel *SquirrelService) getDefStarlarkString(ctx context.Context, node Node) (ret *Node, err error) {
+	sitterQuery, err := sitter.NewQuery([]byte(loadQuery), node.LangSpec.language)
+	if err != nil {
+		return nil, errors.Newf("failed to parse query: %s\n%s", err, loadQuery)
+	}
+	defer sitterQuery.Close()
+	cursor := sitter.NewQueryCursor()
+	defer cursor.Close()
+	cursor.Exec(sitterQuery, getRoot(node.Node))
+
+	for {
+		match, ok := cursor.NextMatch()
+		if !ok {
+			return nil, nil
+		}
+
+		if len(match.Captures) < 3 {
+			panic(match.Captures)
```

Comment:
Number of panics here, can we replace them with breadcrumbs and/or returning errors? Im not exactly sure on the convention Chris has in place

Reply:
You can return an error here pretty easily:

```suggestion
			return nil, errors.Newf("expected 3 captures in starlark query, got %d", len(match.Captures))
```